### PR TITLE
GAN and Autoencoder tutorial notebooks

### PR DESCRIPTION
**Pull request by F20170604**
GAN: https://github.com/tensorflow/swift-models/blob/master/Autoencoder/main.swift
Autoencoder: https://github.com/tensorflow/swift-models/blob/master/Autoencoder/main.swift

I am creating an absolutely new notebook for DCGAN (other than the GAN notebook mentioned in this PR) which will act as the swap with [Python DCGAN](https://www.tensorflow.org/beta/tutorials/generative/dcgan) but unfortunately it's taking more time than I thought it would.

### CODE REVIEW

**Comment by F20170604** [comment]
@rxwei any updates regarding these notebooks?

**Comment by rxwei** [comment]
I'll take a closer look tonight. Sorry for the lack of responses!

**Comment by BradLarson** [comment]
The diff seemed to put everything in one line, so I'll place my Autoencoder comments here:

"by training a Simple Autoencoder" -> I'd lowercase the last two words.

"Define Image and Training Loop parameters along with Image Plot function to save and plot generated images." -> I'd split the code here into two cells, one setting up parameters, and the other defining the plotting function. Additionally, none of the nouns here should be uppercased.

"Notice the relu activation for each layer, except the output layer which uses tanh." -> relu is code-formatted, but tanh is not.

"Define the model and optimizer" -> should be "Define the dataset, model, and optimizer"

The // Training loop comment is unnecessary, given the title.

Should there be a conclusion section at the end after the training run?

**Comment by BradLarson** [comment]
For the GAN notebook:

The two comments of 
```
// Use the AGG renderer for saving images to disk.
//matplotlib.use("Agg")
```
can be removed.

"Notice the LeakyReLU activation for each layer, except the output layer which uses tanh." -> again, I don't think tanh has code formatting here.

"The discriminator is a Dense layer based image classifier." -> "The discriminator uses a series of fully-connected layers to form an image classifier."

Remove the // Models comment.

"Define loss functions and optimizers for both models." -> This is probably unnecessary.

The code for the two loss functions should be split into the two sections. Remove the // Loss functions comment.

**Comment by F20170604** [comment]
@BradLarson Updated both the notebooks.

**Comment by BradLarson** [comment]
In the Autoencoder tutorial:

"Define image and Training Loop parameters." -> "Define image and training loop parameters."

"Define image plot function to save and plot generated images." -> "Define a function to display the generated images."

Also, I'd remove the remaining image-saving code, like `plt.savefig`

Remove the comment `/// An autoencoder.`

For the conclusion, it looks like the first two paragraphs and bullet points were copied from https://towardsdatascience.com/autoencoders-bits-and-bytes-of-deep-learning-eaba376f23ad and https://www.jeremyjordan.me/autoencoders/ . We really need these words to be either your own, or drawn from existing TensorFlow materials. Wording in these notebooks cannot be copied from non-TensorFlow sources.

**Comment by Ayush517** [comment]
> For the conclusion, it looks like the first two paragraphs and bullet points were copied from https://towardsdatascience.com/autoencoders-bits-and-bytes-of-deep-learning-eaba376f23ad and https://www.jeremyjordan.me/autoencoders/ . We really need these words to be either your own, or drawn from existing TensorFlow materials. Wording in these notebooks cannot be copied from non-TensorFlow sources.

Oh I didn't knew that. Sorry about this, I'll remove them completely and write them on my own.

**Comment by BradLarson** [comment]
In the GAN tutorial:

"Define image and Training Loop parameters." -> "Define image and training loop parameters."

"Define image plot function to save and plot generated images." -> "Define a function to display the generated images."

As with the autoencoder tutorial, I'd remove the remaining image-saving code, like plt.savefig().

Move the discriminator loss function code into the "Descriminator loss" section.

**Comment by F20170604** [comment]
@Brad done with the notebooks. I've updated the conclusion part in the Autoencoder too. Can you please check.

**Review by BradLarson — APPROVED** [review]
The last two changes I'd make:

Autoencoder:

"Define image and training Loop parameters." -> "Define the image and training loop parameters."

GAN:

"Define image and training Loop parameters." -> "Define the image and training loop parameters."

Then these should be good to pull in.
